components/org.wso2.carbon.identity.fraud.detection.sift/src/main/java/org/wso2/carbon/identity/fraud/detection/sift/SiftConnectorConfig.java

```diff
@@ -0,0 +1,121 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.fraud.detection.sift;
+
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_CATEGORY;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_FRIENDLY_NAME;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_NAME;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_ORDER;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_SUB_CATEGORY;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.SIFT_API_KEY_PROP;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.SIFT_API_KEY_PROP_DESC;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.SIFT_API_KEY_PROP_NAME;
+
+/**
+ * Sift Config Connector containing the configurations required for integrating with Sift.
+ */
+public class SiftConnectorConfig implements IdentityConnectorConfig {
+
+    @Override
+    public String getName() {
+
+        return CONNECTOR_NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+
+        return CONNECTOR_FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getCategory() {
+
+        return CONNECTOR_CATEGORY;
+    }
+
+    @Override
+    public String getSubCategory() {
+
+        return CONNECTOR_SUB_CATEGORY;
+    }
+
+    @Override
+    public int getOrder() {
+
+        return CONNECTOR_ORDER;
+    }
+
+    @Override
+    public Map<String, String> getPropertyNameMapping() {
+
+        Map<String, String> mapping = new HashMap<>();
+        mapping.put(SIFT_API_KEY_PROP, SIFT_API_KEY_PROP_NAME);
+        return mapping;
+    }
+
+    @Override
+    public Map<String, String> getPropertyDescriptionMapping() {
+
+        Map<String, String> mapping = new HashMap<>();
+        mapping.put(SIFT_API_KEY_PROP, SIFT_API_KEY_PROP_DESC);
+        return mapping;
+    }
+
+    @Override
+    public String[] getPropertyNames() {
+
+        List<String> properties = new ArrayList<>();
+        properties.add(SIFT_API_KEY_PROP);
+        return properties.toArray(new String[0]);
+    }
+
+    @Override
+    public Properties getDefaultPropertyValues(String tenantDomain) {
+
+        Map<String, String> defaultProperties = new HashMap<>();
+
+        defaultProperties.put(SIFT_API_KEY_PROP, "");
+        Properties properties = new Properties();
+        properties.putAll(defaultProperties);
+        return properties;
+    }
+
+    @Override
+    public Map<String, String> getDefaultPropertyValues(String[] propertyNames, String tenantDomain) {
+
+        return new HashMap<>();
```

Comment:
is returning empty hashmap expected?

Reply:
This is a practice we have followed in other cases as well. https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.password.expiry/src/main/java/org/wso2/carbon/identity/password/expiry/PasswordExpiryConfigImpl.java#L147

Reply:
Can you explain the purpose of this practice? just asking for educational purposes :)